doc/developers/advanced_installation.rst

```diff
@@ -22,127 +33,191 @@ basis to help users test bleeding edge features or bug fixes::
+Windows
+-------
 
-    every time the source code of a compiled extension is changed (for
-    instance when switching branches or pulling changes from upstream).
-    Compiled extensions are Cython files (ending in `.pyx` or `.pxd`).
+First, install `Build Tools for Visual Studio 2019
+<https://visualstudio.microsoft.com/downloads/>`_.
 
-On Unix-like systems, you can equivalently type ``make in`` from the
-top-level folder. Have a look at the ``Makefile`` for additional utilities.
+.. warning::
 
-Mac OSX
--------
+    You DO NOT need to install Visual Studio 2019. You only need the "Build
+    Tools for Visual Studio 2019", under "All downloads" -> "Tools for Visual
+    Studio 2019".
 
-The default C compiler, Apple-clang, on Mac OSX does not directly support
-OpenMP. We present two solutions to enable OpenMP support (you need to do only
-one).
+Secondly, find out if you are running 64-bit or 32-bit Python. The building
+command depends on the architecture of the Python interpreter. You can check
+the architecture by running the following in ``cmd`` or ``powershell``
+console::
 
-.. note::
+    python -c "import struct; print(struct.calcsize('P') * 8)"
 
-    First, clean any previously built files in the source folder of
-    scikit-learn::
+For 64-bit Python, configure the build environment with::
+
+    SET DISTUTILS_USE_SDK=1
+    "C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Auxiliary\Build\vcvarsall.bat" x64
 
-        make clean
+Replace ``x64`` by ``x86`` to build for 32-bit Python.
 
-Using conda
-~~~~~~~~~~~
+Please be aware that the path above might be different from user to user. The
+aim is to point to the "vcvarsall.bat" file that will set the necessary
+environment variables in the current command prompt.
 
-One solution is to install another compiler which supports OpenMP. If you use
-the conda package manager, you can install the ``compilers`` meta-package from
-the conda-forge channel, which provides OpenMP-enabled C compilers.
+Finally, build scikit-learn from this command prompt::
+
+    pip install --editable .
+
+macOS
+-----
+
+The default C compiler on macOS, Apple clang (confusingly aliased as
+`/usr/bin/gcc`), does not directly support OpenMP. We present two alternatives
+to enable OpenMP support:
+
+- either install `conda-forge::compilers` with conda;
+
+- or install `libomp` with Homebrew to extend the default Apple clang compiler.
+
+macOS compilers from conda-forge
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If you use the conda package manager, you can install the ``compilers``
+meta-package from the conda-forge channel, which provides OpenMP-enabled C/C++
+compilers based on the llvm toolchain.
 
 It is recommended to use a dedicated conda environment to build scikit-learn
 from source::
 
     conda create -n sklearn-dev python numpy scipy cython joblib pytest \
         conda-forge::compilers conda-forge::llvm-openmp
     conda activate sklearn-dev
+    make clean
```

Comment:
Do we want to advise users to `make clean` in the other instructions?

Reply:
corollary: why is `make clean` needed here and not elsewhere?

Reply:
because on macOS you may have build sklearn once without OpenMP support, then came to the doc to see how to enable OpenMP. In that case you need to clean first.

Reply:
I don't think it's useful for the other cases.